cmd/snap/cmd_quota.go

```diff
@@ -72,57 +90,128 @@ func init() {
 	cmd.hidden = true
 }
 
-func (x *cmdQuota) Execute(args []string) (err error) {
+type cmdSetQuota struct {
+	clientMixin
+
+	// MaxMemory and MemoryMax are mutually exclusive and provided for
+	// convienience, they mean the same.
+	MemoryMax  string `long:"memory" optional:"true"`
+	Parent     string `long:"parent" optional:"true"`
+	Positional struct {
+		GroupName string              `positional-arg-name:"<group-name>" required:"true"`
+		Snaps     []installedSnapName `positional-arg-name:"<snap>" optional:"true"`
+	} `positional-args:"yes"`
+}
+
+func (x *cmdSetQuota) Execute(args []string) (err error) {
 	var maxMemory string
 	switch {
-	case x.MaxMemory != "" && x.MemoryMax != "":
-		return fmt.Errorf("cannot use --max-memory and --memory-max together")
-	case x.MaxMemory != "":
-		maxMemory = x.MaxMemory
 	case x.MemoryMax != "":
 		maxMemory = x.MemoryMax
 	}
 
-	if maxMemory == "" && x.Parent == "" && len(x.Positional.Snaps) == 0 {
-		return x.showQuotaGroupInfo(x.Positional.GroupName)
-	}
+	names := installedSnapNames(x.Positional.Snaps)
 
-	// TODO: update without max-memory (i.e. append snaps operation).
-	// Also how do we remove snaps from a group? Should the default be append?
-	// Do we want a "reset" operation to start from scratch?
-	if maxMemory == "" {
-		return fmt.Errorf("missing required --max-memory argument")
+	// figure out if the group exists or not to make error messages more useful
+	groupExists := false
+	if _, err = x.client.GetQuotaGroup(x.Positional.GroupName); err == nil {
+		groupExists = true
 	}
 
-	mem, err := strutil.ParseByteSize(maxMemory)
-	if err != nil {
-		return err
+	switch {
+	case maxMemory == "" && x.Parent == "" && len(x.Positional.Snaps) == 0:
+		// no snaps were specified, no memory limit was specified, and no parent
+		// was specified, so just the group name was provided - this is not
+		// supported since there is nothing to change/create
+
+		// TODO: we could specialize this error message depending on whether or
+		// not the quota group exists or not
```

Comment:
is this a TODO about the daemon code? because here we are doing what the TODO says

Reply:
sorry I added this TODO, then I implemented it and forgot to delete the TODO